src/redisearch_rs/tools/shadow_tester/src/datasets.rs

```diff
@@ -0,0 +1,127 @@
+    let text = get_1984_text()?;
+
+    let mut part_num = 0;
+    let mut part_len = 0;
+    let mut index_pos = 0;
+    let mut part_text = vec![];
+
+    let mut setup_queries = vec![
+        ("flushall".to_string(), vec![]),
+        // Create the index first so that it is updated as new entries are added. If the index is
+        // created after the entries, then it will update in the background and the test queries
+        // will return inconsistent results based on the partial index.
+        (
+            "FT.CREATE".to_string(),
+            vec![
+                "idx:1984".to_string(),
+                "ON".to_string(),
+                "HASH".to_string(),
+                "PREFIX".to_string(),
+                "1".to_string(),
+                "1984:part:".to_string(),
+                "SCHEMA".to_string(),
+                "text".to_string(),
+                "TEXT".to_string(),
+                "WEIGHT".to_string(),
+                "1.0".to_string(),
+                "part_num".to_string(),
+                "NUMERIC".to_string(),
+                "SORTABLE".to_string(),
+                "index_pos".to_string(),
+                "NUMERIC".to_string(),
+                "SORTABLE".to_string(),
+                "length".to_string(),
+                "NUMERIC".to_string(),
+                "SORTABLE".to_string(),
+            ],
+        ),
+    ];
+
+    for line in text.lines() {
+        if line.is_empty() {
+            let t = part_text.join(" ");
+            setup_queries.push((
+                "HSET".to_string(),
+                vec![
+                    format!("1984:part:{part_num}"),
+                    "text".to_string(),
+                    t,
+                    "part_num".to_string(),
+                    part_num.to_string(),
+                    "index_pos".to_string(),
+                    index_pos.to_string(),
+                    "part_len".to_string(),
+                    part_len.to_string(),
+                ],
+            ));
+
+            part_num += 1;
+            index_pos += part_len + 1;
+            part_text.clear();
+            part_len = 0;
+        } else {
+            part_text.push(line);
+            part_len += line.len() + 1;
+        }
+    }
+
+    let t = part_text.join(" ");
+    setup_queries.push((
+        "HSET".to_string(),
+        vec![
+            format!("1984:part:{part_num}"),
+            "text".to_string(),
+            t,
+            "part_num".to_string(),
+            part_num.to_string(),
+            "index_pos".to_string(),
+            index_pos.to_string(),
+            "part_len".to_string(),
+            part_len.to_string(),
+        ],
+    ));
+
+    Ok(setup_queries)
+}
+
+/// Gets the 1984 book text from the cache if it exists, otherwise fetches it from the Gutenberg
+/// Project and stores it in the cache.
+fn get_1984_text() -> Result<String, ureq::Error> {
+    let cache_path = PathBuf::from("cache/1984.txt");
```

Comment:
This way, the cache path depends on the current directory when the tool is invoked. Is that the intention?

Reply:
No, not really intended. Just used a simpler implementation from the `trie_bencher` to get the file. Do you have a specific static location for the cache folder?